### DISCUSSION TITLE
Search component

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#656] [Feedback]
Use this place to discuss the [search component](https://ons-design-system.netlify.com/components/search) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by matthewandrewpalmer** [comment]
For the Character check Variant of the search component, would it be possible to have a variant that either had a range or just a limit check. 

In the Blaise survey manager, we have a search by serial number which is a minimum of 8 characters but it could be up to 11 characters.

**Comment by boxadesign** [comment]
Yes this wouldn't be a problem. Are you happy to do this work and raise a PR?

**Comment by matthewandrewpalmer** [comment]
@boxadesign I have implemented this with a parameter to only check against that the input is at least the required length and i've updated the documentation. Would be good if you could have a look at it, ~~I can't push or create a PR as I don't have access~~. Made a merge request with this in.

**Comment by trevorsaint** [comment]
Few problems with this component whilst working on the 'News and events' design.

The default usage of the macro results in:

1.  **Empty label attribute** (should be populated with a default label title even if it is hidden using a utility class)

2. **Classes modifier** (should allow flexibility in the width of the input if required. A class is added to the input, however, there is no space between class additions, which causes a bug

We also need to look at the ordering of the classes. The class additions seem to come before the already defined width attribute and therefore would have no effect on the width anyway

3. **Responsive width** (should scale nicely and be full width on mobile. However due to the `flex-flow: row wrap` when the screen is reduced the component breaks. Removing this attribute fixes the issue

4. **Unused classes** (classes have been added that have no effect, and should really be removed)

5. **Enhancements** (adding the option for attributes would be a good improvement)

**Reply by jrbarnes9** [reply]
#764 resolved

**Comment by jrbarnes9** [comment]
Double error border on search with character check

![Screenshot 2021-06-21 at 14 20 39](https://user-images.githubusercontent.com/43346934/122768885-f1988300-d29b-11eb-9da2-4febc21897c9.png)

**Reply by rmccar** [reply]
Fixed in #1543

**Comment by jrbarnes9** [comment]
Search component input height too tall on xs/s bp

![Screenshot 2021-10-04 at 09 52 49](https://user-images.githubusercontent.com/43346934/135822176-be1e6ddb-edef-4319-b0c7-30bff2d217b1.png)

**Reply by jrbarnes9** [reply]
Raised #1714
